components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/mappings/SubscriptionThrottlePolicyMappingUtil.java

```diff
@@ -0,0 +1,157 @@
+                        JSONObject attribute = (JSONObject) attributeObj;
+                        CustomAttributeDTO customAttributeDTO = CommonThrottleMappingUtil.getCustomAttribute(
+                                attribute.get(RestApiConstants.THROTTLING_CUSTOM_ATTRIBUTE_NAME).toString(),
+                                attribute.get(RestApiConstants.THROTTLING_CUSTOM_ATTRIBUTE_VALUE).toString());
+                        customAttributeDTOs.add(customAttributeDTO);
+                    }
+                    policyDTO.setCustomAttributes(customAttributeDTOs);
+                }
+            } else {
+                log.error("policy object " + policy.toString() + " is not aa Subscription Policy, hence will not "
+                        + "be considered");
+            }
+            if (policy.getDefaultQuotaPolicy() != null) {
+                policyDTO.setDefaultLimit(CommonThrottleMappingUtil.fromQuotaPolicyToDTO(policy.getDefaultQuotaPolicy()));
+            }
+            return policyDTO;
+        } catch (ParseException | UnsupportedThrottleLimitTypeException e) {
+            throw new SubscriptionThrottlePolicyException(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Converts a single Subscription Policy DTO into a model object
+     *
+     * @param dto Subscription policy DTO object
+     * @return Converted Subscription policy model object
+     * @throws UnsupportedThrottleLimitTypeException
+     */
+    @SuppressWarnings("unchecked") public static SubscriptionPolicy fromSubscriptionThrottlePolicyDTOToModel(
+            SubscriptionThrottlePolicyDTO dto) throws UnsupportedThrottleLimitTypeException {
+
+        SubscriptionPolicy subscriptionPolicy = new SubscriptionPolicy(dto.getPolicyName());
+        subscriptionPolicy = CommonThrottleMappingUtil.updateFieldsFromDTOToPolicy(dto, subscriptionPolicy);
+        subscriptionPolicy.setBillingPlan(dto.getBillingPlan());
+        subscriptionPolicy.setRateLimitTimeUnit(dto.getRateLimitTimeUnit());
+        subscriptionPolicy.setRateLimitCount(dto.getRateLimitCount());
+        subscriptionPolicy.setStopOnQuotaReach(dto.getStopOnQuotaReach());
+
+        List<CustomAttributeDTO> customAttributes = dto.getCustomAttributes();
+        if (customAttributes != null && customAttributes.size() > 0) {
+            JSONArray customAttrJsonArray = new JSONArray();
+            for (CustomAttributeDTO customAttributeDTO : customAttributes) {
+                JSONObject attrJsonObj = new JSONObject();
+                attrJsonObj.put(RestApiConstants.THROTTLING_CUSTOM_ATTRIBUTE_NAME, customAttributeDTO.getName());
+                attrJsonObj.put(RestApiConstants.THROTTLING_CUSTOM_ATTRIBUTE_VALUE, customAttributeDTO.getValue());
+                customAttrJsonArray.add(attrJsonObj);
+            }
+            subscriptionPolicy.setCustomAttributes(customAttrJsonArray.toJSONString().getBytes(StandardCharsets.UTF_8));
```

Comment:
Fixed in https://github.com/wso2/carbon-apimgt/commit/8bf295d69103fd9391adef96f70d05a766cb4879.